### DISCUSSION TITLE
Wire:change Event in SelectAction

### DISCUSSION THREAD

**Discussion by gotzsys** [filamentphp/filament#8741] [Help] [answered]
Hello everyone,
Is there a way to trigger the wire:change event when selecting an item from a SelectAction? Or do I need to create a custom component?"

```
protected function getHeaderActions(): array
{
    return [
        SelectAction::make('changeCategory')
            ->options(ReviewCategory::all()->pluck('name', 'id')->toArray())
            ->action(fn ($category) => $this->setStatusFilter($category))
    ];
}
```

**Comment by leandrocfe** [comment, marked as answer]
```php
public $changeCategory = null;

protected function getHeaderActions(): array
{
    return [
        SelectAction::make('changeCategory')
        ->options(ReviewCategory::pluck('name', 'id'))
    ];
}

public function updatedChangeCategory()
{
    $this->setStatusFilter($this->changeCategory);
}
```

**Reply by gotzsys** [reply]
"@leandrocfe, you're awesome, it worked perfectly. Let me ask one more question. This selectAction will be a category filter; I don't want to use the native filter because I can't allow the user to remove the filter; it should only be viewed by category. Is it possible to access the selected ID value there in the resource in the getEloquentQuery method?

```
public static function getEloquentQuery(): Builder {
    return parent::getEloquentQuery()->where('category_id', 'fb42c692-eeac-4e37-8e60-548b45c53a8f');
}
```

**Reply by leandrocfe** [reply]
You could do this on the ListPage

```php
protected function getTableQuery(): ?Builder
{
    return parent::getTableQuery()->whereCategoryId($this->changeCategory);
}
```

or via [modifyQueryUsing](https://filamentphp.com/docs/3.x/panels/resources/listing-records#customizing-the-table-eloquent-query)

**Reply by abdulmannans-joistic** [reply]
@leandrocfe What to do if I have multiple Select Action and they are dependent on each other. How can i get selected option value of the first one

**Reply by jimiero** [reply]
@abdulmannans-joistic have you ever found a solution?

**Reply by abdulmannans-joistic** [reply]
> @abdulmannans-joistic have you ever found a solution?

I don't remember but it was not the permanent fix.
